void_migration/plotter.py

```diff
@@ -103,20 +103,31 @@ def plot_s_bar(y, s_bar, nu_time, p):
 def plot_s(x, y, s, p, t):
     plt.figure(fig)
     plt.clf()
+    ##################### if charge_discharge and mono-disperse #####################################
+    colors = [(1, 0, 0), (0, 0, 1)]
+    cmap_name = 'my_list'
+    cmap = LinearSegmentedColormap.from_list(cmap_name, colors, N=2)
+    #################################################################################################
     with warnings.catch_warnings():
         warnings.simplefilter("ignore", category=RuntimeWarning)
         s_plot = np.nanmean(s, axis=2).T
     s_plot = np.ma.masked_where(np.isnan(s_plot), s_plot)
-    plt.pcolormesh(x, y, s_plot, cmap=orange_blue_cmap, vmin=0.001, vmax=0.01)
+    # plt.pcolormesh(x, y, s_plot, cmap=orange_blue_cmap, vmin=p.s_m, vmax=p.s_M)
     # plt.colorbar()
+    if p.charge_discharge:
+        # plt.pcolormesh(x, y, s_plot, cmap=orange_blue_cmap, vmin=p.s_m, vmax=p.s_M)
+        plt.pcolormesh(x, y, s_plot, cmap=cmap, vmin=0.000045, vmax=0.00004505)
```

Comment:
we cant have hardcoded values like this here. can you calculate them from the input parameter file instead?